Update core analysis docs

cbethell opened the pull request:
**Issue Addressed**
Closes #275, #274, and #263

**What is the purpose of these changes?**
<!--Provide some background on the changes proposed.-->
This PR makes the recommended updates to the main `README.md` file in preparation for further usability evals.

**What changes did you make?**
<!--Describe the concrete changes that you made or additional features that were added, be as detailed as possible in the steps you took.-->
The following changes are made in this PR:
- the main steps of setting up and running the workflow are numbered for easier interpretation (#275)
- emphasis is placed on the parts of the example commands that users would need to replace (#274)
- the minimum Snakemake version is noted as determined by tracking the changelog and the [PR filed](https://github.com/snakemake/snakemake/pull/493) to update Snakemake's documentation (#263)
**Were there any other solutions that you tried?**
<!--Did you try alternative solutions that did or didn't work before choosing this one? Why did you choose this route?-->

**Any comments, concerns, or questions important for reviewers**

Do these changes seem to adequately address the issues mentioned/tagged above?

**Checklist**

Place an `x` in all boxes that you have completed.

- [ ] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

allyhawkins commented on README.md:
```diff
@@ -63,8 +63,8 @@ The workflow can directly take as input the `filtered` RDS files downloaded from
   --use-conda \
-  --config results_dir="relative path to relevant results directory" \
-  project_metadata="relative path to your-project-metadata.TSV"
+  --config results_dir="<REPLACE_WITH_RELATIVE_PATH_TO_RESULTS_DIRECTORY>" \
+  project_metadata="<REPLACE_WITH_RELATIVE_PATH_TO_YOUR_PROJECT_METADATA_TSV>"
```
I actually think I would remove the underscores in between the words. I'm also not crazy about the all caps, but maybe that will draw people's attention to it. 

I also think you want to add a few sentences below that state: "Where `config_results_dir` is the relative path to the results directory where all results from running the workflow will be stored". And then do the same for the project metadata. 
We do something similar in `scpca-nf` - https://github.com/AlexsLemonade/scpca-nf/blob/main/external-instructions.md#configuring-scpca-nf-for-your-environment

allyhawkins commented on README.md:
```diff
@@ -100,15 +100,16 @@ More instructions on cloning a GitHub repository can be found [here](https://doc
 
 The core downstream single-cell analysis pipeline, which includes filtering, normalization, dimensionality reduction, and clustering is implemented using a Snakemake workflow.
 Therefore, you will also need to install Snakemake before running the pipeline.
+Note that the **minimum** version of Snakemake you will need installed to be compatible with conda is version **5.23.0**.
```
```suggestion
Note that the **minimum** version of Snakemake you will need to have installed to be compatible with conda is version **5.23.0**.
```

allyhawkins commented on README.md:
```diff
@@ -209,9 +210,9 @@ The below code is an example of running the Snakemake workflow using the project
-  --config results_dir="relative path to relevant results directory" \
-  project_metadata="relative path to your-project-metadata.TSV" \
-  mito_file="full path to your-mito-file.txt"
+  --config results_dir="<REPLACE_WITH_RELATIVE_PATH_TO_RESULTS_DIRECTORY>" \
```
Looking at the fact that we have this multiple places, I wonder if it would actually be better to include a bolded note under the command reminding users that they will need to replace the paths for `results_dir` and `project_metadata`. And then I would make the following change instead: 
```suggestion
  --config results_dir="<RELATIVE PATH TO RESULTS DIRECTORY>" \
```

allyhawkins reviewed (commented):
I think rather than changing to the all caps, underscore separated paths in the examples, we should include notes about what the individual arguments represent and then a reminder to replace the paths in bold immediately under the command. I think by explicitly stating it we might get through to more people and we should be sure to explain what each option actually represents.

cbethell commented:
I've implemented your suggestions to add a line or two explaining the parameters and bolding/emphasizing the need to replace with the relevant filepaths @allyhawkins -- I did keep the CAPS since that was an original suggestion from Deepa but agree that the underscores were much so I removed them. Let me know what you think now!

allyhawkins commented on README.md:
```diff
@@ -63,10 +63,14 @@ The workflow can directly take as input the `filtered` RDS files downloaded from
 
+Where `results_dir` is the relative path to the results directory where all results from running the workflow will be stored and `project_metadata` is the relative path to the TSV file containing the relevant information about your input files.
+See more information on project metadata in [section 3](#3-metadata-file-format) below.
+**You will want to replace the paths for both `results_dir` and `project_metadata` to successfully run the workflow.** 
```
I think we want to switch the order of these sentences so the warning comes first.

allyhawkins reviewed (approved):
This looks good! Just one sentence order switch, but no need for me to look at it again.
